### DISCUSSION TITLE
Have model get $connection from default config if none specified

### DISCUSSION THREAD

**Discussion by CountTo25** [laravel/framework#39262] [Ideas]
When working together with legacy i found weird issue (or more like edge case?)

Using $model->fresh(['relation']) when relation uses different connection assumes that relation has same connection as $model, unless declared otherwise. 
Even if relation uses default db, $model with non-default connection will freak out when using ->fresh(), since it will try to use its custom connection specified in model.

Probably having model boot its $connection with config('database.default') would solve it, but i have no idea on how much of a strain on performance that will add. I've tested changing stuff locally and found that updating model class to have

```php
public function __construct(array $attributes = [])
{
    if ($this-connection === null) {$this->connection = config('database.default')
    ...
}
```

but i'm too dumb to commint my own PR to something as big as laravel, so i decided to discuss this here first

Here's how to replicate

```php
Class Book extends Model {
  protected $connection = 'db_legacy';
  
  public function authors(): HasMany {
    return $this->hasMany(Author::class)
  }
}
//---------------------
Class Author extends Model {

}
//---------------------
$book = Book::first();
$book->fresh(['authors']); //UH OH
```

```php
Class Book extends Model {
  protected $connection = 'db_legacy';
  
  public function authors(): HasMany {
    return $this->hasMany(Author::class)
  }
}
//---------------------
Class Author extends Model {
  protected $connection = 'mysql';
}
//---------------------
$book = Book::first();
$book->fresh(['authors']); //OK
```

**Comment by linaspasv** [comment]
Hi, I have just experienced similar issue where Eloquent model without pre-defined connection where it tries to use one from the parent. I think it's completely wrong. It would be awesome if you could create a PR to this. Thanks.

**Reply by CountTo25** [reply]
> Hi, I have just experienced similar issue where Eloquent model without pre-defined connection where it tries to use one from the parent. I think it's completely wrong. It would be awesome if you could create a PR to this. Thanks.

Well, as i expected, that was a bad idea. While i know general idea of resolving this, as Taylor said, calling config out of eloquent is bad idea and seems only current workaround is specifying $connection on every model so you dont get bonked with this.

**Reply by linaspasv** [reply]
> > Hi, I have just experienced similar issue where Eloquent model without pre-defined connection where it tries to use one from the parent. I think it's completely wrong. It would be awesome if you could create a PR to this. Thanks.
> 
> Well, as i expected, that was a bad idea. While i know general idea of resolving this, as Taylor said, calling config out of eloquent is bad idea and seems only current workaround is specifying $connection on every model so you dont get bonked with this.

My workaround to this is to have a custom Model class with defined default connection at the moment.

```php
<?php

namespace App\Foundation\Database\Eloquent;

use Illuminate\Database\Eloquent\Model as BaseModel;

abstract class Model extends BaseModel
{
    protected $connection = 'mysql';
}
```

and then I have alias in ``config/app.php`` for Eloquent that I use to extend all of my models

```php
'Eloquent' => App\Foundation\Database\Eloquent\Model::class,
```

It's a shame that these things are not out of the box but it is what it is :-)
